screens/Meal.tsx

```diff
@@ -1,210 +1,303 @@
+import AsyncStorage from '@react-native-async-storage/async-storage';
 import {BottomTabScreenProps} from '@react-navigation/bottom-tabs';
-import React, {useState} from 'react';
+import React, {useState, useEffect} from 'react';
 import {StyleSheet} from 'react-native';
 import {RootTabList} from '../App';
-import {Box, Center, HStack, ScrollView, Text, VStack} from 'native-base';
-import _ from 'lodash';
+import {
+  Box,
+  Center,
+  HStack,
+  ScrollView,
+  Text,
+  VStack,
+  Modal,
+  Button,
+} from 'native-base';
+import _, {size} from 'lodash';
+import axios from 'axios';
+import {parse} from 'node-html-parser';
 
 type Props = BottomTabScreenProps<RootTabList, 'Meal'>;
 
 export default function Meal({navigation}: Props) {
-  // todo
-  //   const favoritedNumbers = DUMMY_MEAL_INFO.map(({isFavorited}) =>
-  //     Number(isFavorited),
-  //   ).reduce((sum, currValue) => sum + currValue);
-  //   const notFavoritedNumbers = DUMMY_MEAL_INFO.length - favoritedNumbers;
+  type TodaysMenu = {
+    [name: string]: {
+      breakfast: string;
+      lunch: string;
+      dinner: string;
+      contact: string;
+    };
+  };
+  // state 선언
+  const [menu, setMenu] = useState<TodaysMenu | null>(null); // store menu data here
+  const [isFavorite, setIsFavorite] = useState<{} | null>(null); // store favorite or not here
+  const [selectedMeal, setSelectedMeal] = useState<string | null>(null); // store selected meal (modal) here
+
+  // 식단 정보 크롤링
+  const exampleDateURL =
+    'https://snuco.snu.ac.kr/ko/foodmenu?field_menu_date_value_1%5Bvalue%5D%5Bdate%5D=&field_menu_date_value%5Bvalue%5D%5Bdate%5D=09%2F23%2F2021';
+  const todayMenuURL = 'https://snuco.snu.ac.kr/ko/foodmenu'; // 정보 받아와서 리스트에 저장, 각 식당의 리스트 인덱스 따로 저장
+
+  useEffect(() => {
+    axios.get(exampleDateURL).then(res => {
+      const html = res.data;
+      const root = parse(html);
+      const data: TodaysMenu = {};
+      _.chain(root.querySelector('tbody').childNodes)
+        .map(trNode => {
+          const trTexts = _.chain(trNode.childNodes)
+            .map((tdNode, idx) =>
+              tdNode.innerText
+                .split(/\s|\t|\n/)
+                .filter(item => item.length > 0)
+                .join(' '),
+            )
+            .value();
+          return trTexts;
+        })
+        .filter(trTexts => trTexts.length > 0)
+        .forEach((trTexts, idx) => {
+          const [a, nameAndContact, c, breakfast, e, lunch, g, dinner] =
+            trTexts;
+          const splitedNameAndContact = _.split(nameAndContact, '(');
+          const name = splitedNameAndContact[0];
+          const contact = splitedNameAndContact[1].slice(0, -1);
+          data[name] = {breakfast, lunch, dinner, contact};
+        })
+        .value();
```

Comment:
[Optional] 좀 더 간결한 코드
```suggestion
      const data: TodaysMenu = _.chain(root.querySelector('tbody').childNodes)
        .map(trNode => {
          const trTexts = _.chain(trNode.childNodes)
            .map((tdNode, idx) =>
              tdNode.innerText
                .split(/\s|\t|\n/)
                .filter(item => item.length > 0)
                .join(' '),
            )
            .value();
          return trTexts;
        })
        .filter(trTexts => trTexts.length > 0)
        .map((trTexts, idx) => {
          const [nameAndContact, breakfast, lunch, dinner] = trTexts.filter(
            trText => trText.length > 0,
          );
          const [name, contact] = nameAndContact.split(/\(|\)/);
          return {
            name,
            breakfast,
            lunch,
            dinner,
            contact,
          };
        })
        .keyBy(({name}) => name)
        .value();
```

Reply:
이대로 반영하니까 안 돌아가서 일단
`const [name, contact] = nameAndContact.split(/\(|\)/);`
이런거 다듬는거만 반영하겠음